addons/pos_restaurant/static/src/js/Screens/SplitBillScreen/SplitBillScreen.js

```diff
@@ -96,29 +96,11 @@ odoo.define('pos_restaurant.SplitBillScreen', function(require) {
         _splitQuantity(line) {
             const split = this.splitlines[line.id];
 
```

Comment:
Maybe I would see something like this
```suggestion
        _splitQuantity(line) {
            const split = this.splitlines[line.id];
            
            const lineQty = line.get_quantity();
            if(lineQty > 0) {
                if (!line.get_unit().is_pos_groupable) {
                    if (split.quantity !== lineQty) {
                        split.quantity = lineQty;
                    } else {
                        split.quantity = 0;
                    }
                } else {
                    if (split.quantity < lineQty) {
                        split.quantity += line.get_unit().is_pos_groupable? 1: line.get_unit().rounding;
                        if (split.quantity > lineQty) {
                            split.quantity = lineQty;
                        }
                    } else {
                        split.quantity = 0;
                    }
                }
            }
        }
```